include/alpaka/block/shared/dyn/BlockSharedMemDynMember.hpp

```diff
@@ -78,20 +79,29 @@ namespace alpaka
             return m_mem.data() + m_dynPitch;
         }
 
-        /*! \return the remaining capacity for static block shared memory.
+        /*! \return the remaining capacity for static block shared memory,
+                    returns a 32-bit type for register efficiency on GPUs
             */
-        std::size_t staticMemCapacity() const
+        std::uint32_t staticMemCapacity() const
         {
             return staticAllocBytes() - m_dynPitch;
         }
 
         //! Storage size in bytes
-        static constexpr std::size_t staticAllocBytes() {return detail::BlockSharedMemDynMemberStatic<TStaticAllocKiB>::staticAllocBytes;}
+        //! Returns a 32-bit type for register efficiency on GPUs
```

Comment:
Should be `\return` instead of prose "Returns" for Dxygen, even though it does not actually describe what the meaning of its return value is. Maybe it should do both, though.
```c++
//! \returns size of statically allocated memory available for both dynamic and static shared memory. Value is of a 32-bit type for register efficiency on GPUs

Reply:
Fixed and rebased.